### PR TITLE
generator: cs: add C# generator

### PR DESCRIPTION
**Pull request by alarsyo**

### CODE REVIEW

File: tools/generator/filters/__init__.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def load_library_in(env):
         common,
         c,
         caml,
+        cs,
```
Comment (seirl) [inline tools/generator/filters/__init__.py:38]: csharp

File: tools/generator/filters/cs.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+from functools import partial
+
+from . import register_filter
+from .common import camel_case, generic_comment, get_array_inner, is_array
+
+
+def cs_type(type) -> str:
```
Comment (seirl) [inline tools/generator/filters/cs.py]: Use `csharp_` prefix everywhere

File: tools/generator/templates/player/cs/Makefile-cs.jinja2 [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2020 Association Prologin <association@prologin.org>
+#}
+{% extends "includes/Makefile-language.jinja2" %}
+
+{% block lang %}cs{% endblock %}
```
Comment (alarsyo) [inline tools/generator/templates/player/cs/Makefile-cs.jinja2:6]: @seirl should I switch to `csharp` here as well ? Not sure what stechec expects from the content of the `_lang` file, it was `cs` for the ruby generator

Comment (alarsyo) [reply tools/generator/templates/player/cs/Makefile-cs.jinja2:6]: let's stick to `cs` for now

File: tools/generator/templates/player/cs/Makefile-cs.jinja2 [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+{% block extra_flags %}
+MCS_FLAGS = -target:library -nowarn:0169,0649
+CXXFLAGS += $(shell pkg-config --cflags glib-2.0 mono-2)
+LDFLAGS = -lm $(shell pkg-config --libs glib-2.0 mono-2)
+{% endblock %}
+
+{% block all_dependencies %}champion.so champion.dll{% endblock %}
```
Comment (seirl) [inline tools/generator/templates/player/cs/Makefile-cs.jinja2]: Use `{{ super() }}`

**Review by seirl — APPROVED** [review]

**Pull request merged**